googleapis_auth/lib/src/client_with_headers.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:http/http.dart';
+
+import 'http_client_base.dart';
+
+/// Returns a [Client] which adds [headers] to the request headers of every
+/// request.
+///
+/// Note: the returned [Client] modifies the [BaseRequest] passed to
+/// [Client.send] by adding [headers]. Any existing headers with the same
+/// key are overwritten.
+Client clientUsingHeaders(
```

Comment:
Are we sure this belongs this package.. this seems more like a package:http specific utility..